src/wiktextract/extractor/de/gloss.py

```diff
@@ -3,58 +3,167 @@
                     sortid="extractor/de/glosses/extract_glosses/28",
                 )
 
-            gloss_data["glosses"] = [gloss_text]
+            gloss_text = extract_tags_from_gloss_text(gloss_data, gloss_text)
 
-            page_data[-1]["senses"].append(gloss_data)
+            if gloss_text or not sub_glosses_list_nodes:
+                gloss_data["glosses"] = [gloss_text]
+                page_data[-1]["senses"].append(gloss_data)
+
+            for sub_list_node in sub_glosses_list_nodes:
+                process_gloss_list_item(
+                    wxr,
+                    page_data,
+                    sub_list_node,
+                    senseid,
+                    gloss_data if not gloss_text else None,
+                )
 
         else:
             wxr.wtp.debug(
                 f"Unexpected list item in glosses: {list_item_node}",
                 sortid="extractor/de/glosses/extract_glosses/29",
             )
             continue
+
+
+def handle_sense_modifier(wxr, list_item_node):
+    wxr.wtp.debug(
+        f"Skipped a sense modifier in gloss list: {list_item_node}",
+        sortid="extractor/de/glosses/extract_glosses/19",
+    )
+    # XXX: We should extract the modifier. However, it seems to affect
+    # multiple glosses. Needs investigation.
+    pass
+
+
+def process_K_template(
+    wxr: WiktextractContext,
+    gloss_data: defaultdict(list),
+    list_item_node: NodeKind.LIST_ITEM,
+) -> None:
+    for template_node in list_item_node.find_child(NodeKind.TEMPLATE):
+        if template_node.template_name == "K":
+            categories = []
+
+            temp_node = copy.deepcopy(template_node)
+            for key, value in template_node.template_parameters.items():
+                if isinstance(key, int):
+                    temp_node.largs = temp_node.largs[:1] + [[value]]
+                    # Cleaned K template will always end with ":". Remove it.
+                    category = clean_node(wxr, {}, temp_node)[:-1]
+                    if category:
+                        categories.append(category)
+                if key == "ft":
+                    # ft (free text) is used liberally to modify the context
+                    # template. Sometimes it seems to belong rather to the
+                    # gloss itself. Most of the time it is not useful.
+                    # XXX Treat free text in K templates.
+                    continue
+
+            prep = template_node.template_parameters.get("Prä")
+            case = template_node.template_parameters.get("Kas")
+            category = (prep if prep else "") + (" + " + case if case else "")
+            if category:
+                categories.append(category)
+
+            # Remove the template_node from the children of list_item_node
+            list_item_node.children = [
+                c for c in list_item_node.children if c != template_node
+            ]
+
+            gloss_data["tags"].extend(categories)
```

Comment:
Okay. I see. I could have really saved a lot of time here if I had had a better understanding of `clean_node()` and probably also the intermediate stage between wikicode and plain text that `wxr.wtp.node_to_html()` provides.

The problem now is that I am not quite sure how to properly write the tests. `clean_node()` now requires the templates to be present. 
- For my local tests (just to make sure everything works as I think), I gave it access to my local database but that obviously doesn't work for automatic testing. 
- Mocking seems way too complicated now with everything clean_node does.
- I could just copy the template content to the test file and add it as a page to wxr. But that could get quite out of hand depending how many subpages are needed.

Have you had this scenario before?

Reply:
For tests process expanded text, `Wtp.add_page` could be used to add the template page with expanded plain text or use `mock.patch` to provide a return result of the `clean_node()` function.

If a test needs to process the node tree like parsing a table, `Wtp.add_page` could be used again to add the expanded wikitext table to the template page or patch the `Wtp.node_to_wikitext` method.

Maybe breaking the complex test to some smaller tests could be easier?

Reply:
Okay, if that's an acceptable solution, I have manually added all pages required to be present in Wtp.

I don't think the test `test_process_K_template` itself is complicated. It might seem complicated since I added all test_cases in this one test function with `subTest()`. But each single test case is quite straightforward.

 The problem is that the K template itself is complicated (you see how much I had to add to make it work). So if we want to test at least all the examples from the K template documentation (as I do at the moment), we would have to mock quite a lot. Best to just use the original K template definition then.

Reply:
I simplified it anyway.

Ready to merge?